lib/include/jxl/decode.h

```diff
@@ -913,6 +950,142 @@ JXL_EXPORT JxlDecoderStatus JxlDecoderSetJPEGBuffer(JxlDecoder* dec,
+ */
+JXL_EXPORT JxlDecoderStatus JxlDecoderGetBoxType(JxlDecoder* dec, char type[5]);
+
+/**
+ * Outputs whether the given type equals the type returned by
+ * JxlDecoderGetBoxType.
+ *
+ * @param dec decoder object
+ * @param type box type to compare with
+ * @return JXL_TRUE if the box type is available and equal to the given type,
+ * JXL_FALSE otherwise.
+ */
+JXL_EXPORT JXL_BOOL JxlDecoderBoxTypeEquals(JxlDecoder* dec, const char* type);
+
+/**
+ * Outputs the type of the current box, after a JXL_DEC_BOX event occured, as 4
+ * characters plus a null termination character. If the box is of type "brob",
+ * the underlying uncompressed box type is returned instead, not "brob".
+ *
+ * @param dec decoder object
+ * @param type buffer to copy the type into
+ * @return JXL_DEC_SUCCESS if the value is available, JXL_DEC_ERROR if not, for
+ *    example the JXL file does not use the container format.
+ */
+JXL_EXPORT JxlDecoderStatus JxlDecoderGetBoxTypeRaw(JxlDecoder* dec,
+                                                    char type[5]);
+
+/**
+ * Outputs whether the given type equals the type returned by
+ * JxlDecoderGetBoxTypeRaw.
+ *
+ * @param dec decoder object
+ * @param type box type to compare with
+ * @return JXL_TRUE if the raw box type is available and equal to the given
+ * type, JXL_FALSE otherwise.
+ */
+JXL_EXPORT JXL_BOOL JxlDecoderBoxTypeRawEquals(JxlDecoder* dec,
+                                               const char* type);
+
+/**
+ * Outputs whether the raw box type equals "brob", indicating it is a brotli
+ * compressed box. Its underlying type is still returned with
+ * JxlDecoderGetBoxType. The data returned to the box buffer is transparently
+ * decompressed, unless JxlDecoderSetBoxModeRaw is enabled, so this function is
+ * not needed to just get the data, as long as the library is compiled with
+ * brotli.
+ *
+ * @param dec decoder object
+ * @param type box type to compare with
+ * @return JXL_TRUE if the box type is available and is "brob",
+ * JXL_FALSE otherwise.
+ */
+JXL_EXPORT JXL_BOOL JxlDecoderIsBoxCompressed(JxlDecoder* dec,
+                                              const char* type);
+
+/**
+ * Returns the compressed size of a box, after the JXL_DEC_BOX event. For a
+ * non-compressed box, this is the size of the contents. For a compressed brob
+ * box, this is the size of the compressed box contents plus the additional 4
+ * byte indicating the underlying box type. This function is only useful when
+ * getting boxes in raw mode, in the default mode where brob boxes are
+ * uncompressed, this does not return the size of the uncompressed contents,
+ * and there is no way to get the uncompressed size before having already
+ * uncompressed and output this data.
+ *
+ * @param dec decoder object
+ * @param size raw size of the box in bytes
+ * @return JXL_DEC_ERROR if no box size is available, JXL_DEC_SUCCESS otherwise.
+ */
+JXL_EXPORT JxlDecoderStatus JxlDecoderGetBoxSizeRaw(const JxlDecoder* dec,
```

Comment:
Do we need this function?

Reply:
When doing primitive box operations, such as concatenating boxes and leaving some out, then this helps: you know exactly what size to set on the output buffer, which makes it easier. The size field and type field are the two things you can get from the box header, not allowing to see the size field would disable possible functionality of the box format.

Reply:
I'm not sure what would be an use-case that would be made significantly simpler by having this function. I'd propose we leave this out until somebody asks for it, but I can be convinced otherwise ;)

Reply:
It allows allocating a buffer of the exact correct size, so JXL_DEC_BOX_NEED_MORE_OUTPUT won't occur and you won't allocate more than needed. For quick concatenation operatrions, this can be helpful.